tests/TenantDatabaseManagerTest.php

```diff
@@ -222,6 +223,159 @@
 
     $manager->setConnection('mysql2');
     expect($manager->databaseExists($name))->toBeTrue();
+
+    $manager->deleteDatabase($tenant);
+    expect($manager->databaseExists($name))->toBeFalse();
+});
+
+test('tenant database user can be created when using the permission controller MySQL manager', function () {
+    config([
+        'tenancy.database.managers.mysql' => PermissionControlledMySQLDatabaseManager::class,
+    ]);
+
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    $name = 'foo' . Str::random(8);
+
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+    ]);
+    $dbUser = $tenant->tenancy_db_username;
+
+    /** @var PermissionControlledMySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+    $manager->setConnection('mysql');
+
+    expect($manager->userExists($dbUser))->toBeTrue();
+});
+
+test('tenant database can be created on template tenant connection', function () {
```

Comment:
What is the exact point of this test?

Reply:
This test's point is to ensure that tenant connection can be established successfully when the developer provides the `tenant_template_connection`. 

I don't exactly remember why I added this, but I wanted to make sure `tenant_template_connection` works as expected after adding the hostConnection logic.

Reply:
I'm thinking of removing this test as well. Because there are two other tests that are making use of `tenant_template_connection` then there is no point in this test.

Reply:
So this can be resolved now?

Reply:
Yes.

Reply:
Marking this as resolved now.